lxml-stubs/builder.pyi

```diff
@@ -0,0 +1,30 @@
+from typing import Any, Callable, Mapping, Optional
+
+from .etree import _AnyStr, _Element
+
+class ElementMaker:
+    def __init__(
+        self,
+        typemap: Optional[Mapping[Any, Callable[[_Element, Any], None]]] = ...,
+        namespace: Optional[str] = ...,
+        nsmap: Optional[Mapping[Any, _AnyStr]] = ...,
+        # same signature as etree.Element()
+        makeelement: Optional[Callable[..., _Element]] = ...,
+    ) -> None: ...
+    def __call__(
+        self,
+        tag: str,
+        # Although default ElementMaker only accepts _Element and types
+        # interpretable by default typemap (that is str, CDATA and dict)
+        # as children, typemap can be expanded to make sure item of any
+        # type is accepted.
+        *children: Any,
+        **attrib: _AnyStr,

```

Comment:
Is this notation correct for a mapping?

Reply:
As you suspected, byte string is not accepted on py3, as indiated in my gist link above. I must have misread the source code somewhere. But it is kind of interesting that py2 actually accepts unicode attrib.